src/formatDate.js

```diff
@@ -50,7 +50,67 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newDate = date.split(fromFormat[3]);
+  const year = {
+    from: {
+      start: false,
+      mid: false,
+      end: false,
+      2: false,
+      4: false,
+    },
+  };
+
+  year.to = { ...year.from };
+
+  const format = {
+    from: fromFormat,
+    to: toFormat,
+  };
+  const positions = ['start', 'mid', 'end'];
+
+  for (const pos of positions) {
+    for (const source in format) {
+      const ind = positions.indexOf(pos);
+
+      if (format[`${source}`][ind].includes('Y')) {
+        year[`${source}`][`${pos}`] = true;
+        year[`${source}`][`${format[`${source}`][ind].length}`] = true;
```

Comment:
you don't need to use interpolation. 
`format[source]`
would work too.

Reply:
Would be cleaner too